backends/p4test/midend.cpp

```diff
@@ -97,6 +99,8 @@ MidEnd::MidEnd(CompilerOptions& options) {
         new P4::NestedStructs(&refMap, &typeMap),
         new P4::SimplifySelectList(&refMap, &typeMap),
         new P4::RemoveSelectBooleans(&refMap, &typeMap),
+        new P4::FlattenHeaders(&refMap, &typeMap),
+        new P4::FlattenInterfaceStructs(&refMap, &typeMap),
```

Comment:
It looks like this is affecting all P4-14 programs. It may be better not to call FlattenInterfaceStructs in p4test. I expect that should work just fine.

Reply:
We do not want to change the metadata for all P4-14 programs, so it would be better not tun run FlattenInterfaceStructs. We want to just see the effect of your pass. Can you please delete the second call and rerun the tests with -f? There are too many uninteresting changes to the reference files with this modification.